### ISSUE TITLE
Reliance on $_GET

### ISSUE THREAD

**Issue by victor-david** [hybridauth/hybridauth#1312]
## Bug, feature or question?
Bit of a question.  My server routes everything into index.php like so (nginx):
~~~
location / {
    try_files $uri $uri/ /index.php?$uri;
  }
~~~
Therefore, when my app runs for a url such as **https://example.com/signup/twitter**, $_GET contains _signup/twitter_, which I use to route to the proper controller.

On callback from the provider, the query string **?oauth_token=xxx, etc**. is appended, but it's not in the $_GET var where Hybridauth is looking for it. To work around that, before I start using Hybridauth, I parse  **$_SERVER['REQUEST_URI']** (which contains everything) and use that to populate $_GET - this allows the following to work correctly.

https://github.com/hybridauth/hybridauth/blob/b966db16cd9aacfe58f9fdc75402d6b360ea889c/src/Adapter/OAuth1.php#L221

However, in authenicateFinish(), it's using **filter_input** instead of $_GET to get the callback data

https://github.com/hybridauth/hybridauth/blob/b966db16cd9aacfe58f9fdc75402d6b360ea889c/src/Adapter/OAuth1.php#L283

but **filter_input** doesn't get the data (I'm not sure why) and therefore we throw an exception.

To work around _that_, I changed the lines in authenticateFinish() to grab the values from $_GET instead. **And now it all works.**

### And finally my question
Is there a better way? Am I missing something? I didn't want to change Hybridauth source code, but I found no other way.

### Version and provider
3.7.1

**Comment by chrisgraham** [comment]
Honestly what you're doing seems clunky. PHP should have accurate data for the URL, especially because oAuth is designed to work via injection of GET parameters.

Can you put the path data into an individual GET parameter rather than the query string as a whole, and allow the existing GET parameters to persist? I know Apache redirects have something called "QSA" http://httpd.apache.org/docs/2.4/rewrite/flags.html#flag_qsa
Does nginx have something like that?

**Comment by victor-david** [comment]
Yes, I agree it's clunky. Which is why I was hoping to find a better way. I could put the path data in an individual GET parm, but it would require modfication of the low-level request and router code.

Under Apache, yes, this works with QSA. I've done it before. There might be something I'm missing with nginx, but I tried a few things at the nginx config level and it still didn't work....

Since my first posting, I further modified Hybridauth to use a single point of parm collection (a method in AbstractAdapter) and a optional callback to surface the parm collection to the consumer app.

~~~php
protected function getFilteredInput(int $type, string $varname, $filter = FILTER_DEFAULT, $options = 0)
{
    if ($this->filteredInputCallback != null) {
        return ($this->filteredInputCallback)($type, $varname, $filter, $options);
    }
    return filter_input($type, $varname, $filter, $options);
}
~~~

This lets HA use a consistent way of obtaining parms - instead of some directly from _$_GET_ and some via _filter_input_ - and allows my app to provide the parms without modifying the global $_GET, which I never liked.

Anyway, it's all working, OAuth1 and 2. Meanwhile, I'll investigate nginx more and maybe find another way. 

Thanks for your response.
